x/update_validators/controller.go

```diff
@@ -0,0 +1,48 @@
+package update_validators
+
+import (
+	"github.com/confio/weave"
+	"github.com/confio/weave/orm"
+	abci "github.com/tendermint/abci/types"
+)
+
+// Controller is the functionality needed by
+// cash.Handler and cash.Decorator. BaseController
+// should work plenty fine, but you can add other logic
+// if so desired
+type Controller interface {
```

Comment:
This is good. You can also use a function if you want...
The main point of the controller interface in eg. x/cash was that I wanted to allow people to make use of the business logic, but swap out the storage logic. Or make use of the messages and handlers and add an extra check to the business logic.

Probably over-abstracted fro most needs. So, fine like this. If you feel this is too much abstraction, feel free to make it more concrete

Reply:
I like the idea of everything being similar. That gives a better idea to people who actually compare extensions trying to write a new one.

I'm going to work on the docs after.